src/liballoc/collections/btree/set.rs

```diff
@@ -164,17 +164,33 @@ impl<T: fmt::Debug> fmt::Debug for SymmetricDifference<'_, T> {
 /// [`intersection`]: struct.BTreeSet.html#method.intersection
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct Intersection<'a, T: 'a> {
-    a: Peekable<Iter<'a, T>>,
-    b: Peekable<Iter<'a, T>>,
+    inner: IntersectionInner<'a, T>,
+}
+#[derive(Debug)]
+enum IntersectionInner<'a, T: 'a> {
+    Stitch {
+        a_iter: Iter<'a, T>, // for size_hint, should be the smaller of the sets
+        b_iter: Iter<'a, T>,
+    },
+    Search {
+        a_iter: Iter<'a, T>, // for size_hint, should be the smaller of the sets
```

Comment:
Since the implementation depends on `a_iter` being smaller I think we should choose more descriptive names here for `a` and `b`.

Reply:
Frankly, I hate the name `a` because it's intermingled with the same lifetime identifier, but didn't dare to change them. I'll go for `small` and `large` in the Search case, `small` and `other` in the Stitch case, unless you object.

Reply:
That sounds good!